### PR TITLE
Switches now must declare they are available to JavaScript to be outputted onto page

### LINKED ISSUE #8854: Improve switches
As @stephanfowler mentioned, all switches end up in the source of every single page.
(theguardian.com -> View Page Source -> Search switches)

Some switches however are only for the backend or for facia-tool.
Do you guys think it's a good idea to filter them out? We could add a flag or something to specify that a given switch should be added to the page or not.

### PR DESCRIPTION
**Pull request by robertberry-zz**
We have a stupid number of switches. Please can we kill some. D:

Fixes https://github.com/guardian/frontend/issues/8854

### CODE REVIEW

**Comment by piuccio** [comment]
Brilliant! Why not making it `Option` default to `false`?

**Comment by robertberry** [comment]
HMMMMmmmm ... 

It doesn't need to be an `Option`, you could just provide a default parameter. But I prefer making people think about it and be explicit up front.

**Comment by piuccio** [comment]
fair enough.
If facia-tool keeps working... :+1:

File: common/app/mvt/MultiVariateTesting.scala [hunk 1]
```diff
@@ -45,7 +45,9 @@ case class TestDefinition (
     name,
     description,
     conf.Off,
-    sellByDate)
+    sellByDate,
+    exposeClientSide = false
```
Comment (rich-nguyen) [inline common/app/mvt/MultiVariateTesting.scala]: I think this should be true, there are at most 9 tests, and so far all the server tests have used the switch value. @uplne

**Comment by rich-nguyen** [comment]
:+1:

**Comment by johnduffell** [comment]
:+1: good idea

**Comment by stephanfowler** [comment]
:+1:

File: common/app/conf/switches.scala [hunk 2]
```diff
@@ -395,288 +653,521 @@ object Switches {
...
 
-  val ABMtRec1 = Switch("A/B Tests", "ab-mt-rec1",
+  val ABMtRec1 = Switch(
+    "A/B Tests",
+    "ab-mt-rec1",
     "Viewability results - Recommendation option 1",
-    safeState = Off, sellByDate = new LocalDate(2015, 6, 2)
+    safeState = Off,
+    sellByDate = new LocalDate(2015, 6, 2),
+    exposeClientSide = false
```
Comment (Calanthe) [inline common/app/conf/switches.scala]: this should be true

File: common/app/conf/switches.scala [hunk 3]
```diff
@@ -395,288 +653,521 @@ object Switches {
...
 
-  val ABMtRec2 = Switch("A/B Tests", "ab-mt-rec2",
+  val ABMtRec2 = Switch(
+    "A/B Tests",
+    "ab-mt-rec2",
     "Viewability results - Recommendation option 2",
-    safeState = Off, sellByDate = new LocalDate(2015, 6, 2)
+    safeState = Off,
+    sellByDate = new LocalDate(2015, 6, 2),
+    exposeClientSide = false
```
Comment (Calanthe) [inline common/app/conf/switches.scala]: this should be true

File: common/app/conf/switches.scala [hunk 4]
```diff
@@ -395,288 +653,521 @@ object Switches {
...
 
-  val ABDeferSpacefinder = Switch("A/B Tests", "ab-defer-spacefinder",
+  val ABDeferSpacefinder = Switch(
+    "A/B Tests",
+    "ab-defer-spacefinder",
     "A/B test to defer execution of spacefinder until images and richlinks have been loaded.",
-    safeState = Off, sellByDate = new LocalDate(2015, 6, 2)
+    safeState = Off,
+    sellByDate = new LocalDate(2015, 6, 2),
+    exposeClientSide = false
```
Comment (Calanthe) [inline common/app/conf/switches.scala]: this should be true

File: common/app/conf/switches.scala [hunk 5]
```diff
@@ -77,299 +82,550 @@ object Switches {
...
 
-  val CommercialSwitch = Switch("Commercial", "commercial",
+  val CommercialSwitch = Switch(
+    "Commercial",
+    "commercial",
     "If this switch is OFF, no calls will be made to the ad server. BEWARE!",
-    safeState = On, sellByDate = never
+    safeState = On,
+    sellByDate = never,
+    exposeClientSide = false
```
Comment (uplne) [inline common/app/conf/switches.scala]: This one is checked in javascriptLaterSteps to initialise commercial scripts so it needs to be exposed to the client side

Comment (robertberry) [inline common/app/conf/switches.scala]: Whoops!

File: common/app/conf/switches.scala [hunk 6]
```diff
@@ -395,288 +653,521 @@ object Switches {
...
 
-  val QuizScoresService = Switch("Feature", "quiz-scores-service",
+  val QuizScoresService = Switch(
+    "Feature",
+    "quiz-scores-service",
     "If switched on, the diagnostics server will provide a service to store quiz results in memcached",
-    safeState = Off, sellByDate = new LocalDate(2015, 8, 16))
+    safeState = Off,
+    sellByDate = new LocalDate(2015, 8, 16),
+    exposeClientSide = false
```
Comment (johnduffell) [inline common/app/conf/switches.scala]: looks good :+1:

**Comment by uplne** [comment]
:+1:

**Pull request merged**